deployment/ccip/changeset/cs_rmn_curse_uncurse.go

```diff
@@ -0,0 +1,275 @@
+package changeset
+
+import (
+	"encoding/binary"
+	"fmt"
+	"slices"
+
+	"github.com/pkg/errors"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+	commoncs "github.com/smartcontractkit/chainlink/deployment/common/changeset"
+)
+
+func GlobalCurseSubject() Subject {
+	return Subject{0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01}
+}
+
+type RMNCurseAction struct {
+	ChainSelector  uint64
+	SubjectToCurse Subject
+}
+
+type CurseAction func(e deployment.Environment) []RMNCurseAction
+
+type RMNCurseConfig struct {
+	MCMS         *MCMSConfig
+	CurseActions []CurseAction
+	Reason       string
+}
+
+func (c RMNCurseConfig) Validate(e deployment.Environment) error {
+	state, err := LoadOnchainState(e)
+
+	if err != nil {
+		return errors.Errorf("failed to load onchain state: %v", err)
+	}
+
+	if len(c.CurseActions) == 0 {
+		return errors.Errorf("curse actions are required")
+	}
+
+	if c.Reason == "" {
+		return errors.Errorf("reason is required")
+	}
+
+	validSelectors := e.AllChainSelectors()
+	validSubjects := make([]Subject, 0, len(validSelectors)+1)
+	for _, selector := range validSelectors {
+		validSubjects = append(validSubjects, SelectorToSubject(selector))
+	}
+	validSubjects = append(validSubjects, GlobalCurseSubject())
+
+	for _, curseAction := range c.CurseActions {
+		result := curseAction(e)
+		for _, action := range result {
+			targetChain := e.Chains[action.ChainSelector]
+			targetChainState := state.Chains[action.ChainSelector]
+
+			if err := commoncs.ValidateOwnership(e.GetContext(), c.MCMS != nil, targetChain.DeployerKey.From, targetChainState.Timelock.Address(), targetChainState.RMNRemote); err != nil {
+				return fmt.Errorf("chain %s: %w", targetChain.String(), err)
+			}
+
+			if !slices.Contains(validSelectors, action.ChainSelector) {
+				return errors.Errorf("invalid chain selector %d for chain %s", action.ChainSelector, targetChain.String())
+			}
```

Comment:
Lets use `deployment.IsValidChainSelector` since thats used in the other changeset validations, see https://github.com/smartcontractkit/chainlink/blob/cd3f6b5ccd526d520817dfc15fe7e581d7fc453f/deployment/helpers.go#L182-L191